rollingpin/elasticsearch.py

```diff
@@ -94,52 +94,70 @@ def update_doc(self, id_, updated_fields):
             )
 
     def build_sync_doc(self, sync_info):
+        timestamp_in_milliseconds = int(time.time()) * 1000
+
         # We aren't aware of the commit ID that is being deployed at initial
         # deploy time.  Once the build sync has happened, we have this
         # information, so we go back and update the original document with a
         # list of the sync targets (i.e. commit IDs being deployed).
         sync = ["%s@%s" % (k, v.get('token', '')[:7])
                 for k, v in sync_info.iteritems()]
         return {
+            'timestamps': {
+                'build': {
+                    'sync': timestamp_in_milliseconds,
+                },
+            },
             'sync_targets': ', '.join(sync),
         }
 
     def deploy_start_doc(self):
         timestamp_in_milliseconds = int(time.time()) * 1000
         return {
             'id': self.deploy_name,
+
+            # Once we change the mapping and reindex the data we can remove
+            # this redundant timestamp field.
             'timestamp': timestamp_in_milliseconds,
+
+            'timestamps': {
+                'deploy': {
+                    'start': timestamp_in_milliseconds,
```

Comment:
Let me know what y'all think about this structure.  After doing this implementation, I'm not sure it's the best way to go.  It's kind of strange to have to figure out where to put event-specific information (like `abort_reason` below) as well as the timestamps.  I like the cleanliness of just firing off separate events.  The downside seems to be that we can't really display information from multiple events in Kibana on the same deploy annotation.  So for that, it makes a lot of sense.